EIPS/eip-6046.md

```diff
@@ -0,0 +1,63 @@
+---
+eip: 6046
+title: Replace SELFDESTRUCT with DEACTIVATE
+description: Change SELFDESTRUCT to not delete storage keys and use a special value in the account nonce to signal deactivation
+author: Alex Beregszaszi (@axic)
+discussions-to: https://ethereum-magicians.org/t/almost-self-destructing-selfdestruct-deactivate/11886
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-11-25
+requires: 2681
+---
+
+## Abstract
+
+Change `SELFDESTRUCT` to not delete all storage keys, and to use a special value in the account nonce to signal *decativated* accounts. Because the semantics of revival change (storage keys may exists), we also rename the instruction to `DEACTIVATE`.
+
+## Motivation
+
+The `SELFDESTRUCT` instruction currently has a fixed price, but is unbounded in terms of how many storage/account changes it performs (it needs to delete all keys). This is an outstanding concern for long.
+
+Furthermore with *Verkle trees* accounts will be organised differently: account properties, including storage, will have individual keys, and it will not be possible to traverse and find all used keys. This makes `SELFDESTRUCT` very challenging to support in Verkle trees.
+
+## Specification
+
+1. Change the rules introduced by [EIP-2681](./eip-2681.md) such that regular nonce increase is bound by `2^64-2` instead of `2^64-1`. This applies from genesis.
```

Comment:
This needs more detail.

Reply:
What exactly? Are you suggesting to duplicate the content of EIP-2681 here?

Reply:
I meant to say this needs to be more formally specified. Right now, "bounded" is unclear. Here's my interpretation of it:
```suggestion
1. If an account's nonce (defined in [EIP-2681](./eip-2681.md)) would be set to a number greater than `2^64-2`, it is set to `2^64-1` instead.
```
"It applies from genesis" is also unclear to me (do you mean starting from when this EIP is activated?), although that is probably a lack of knowledge on my part.

Reply:
EIP-2681 *applies from genesis*, where genesis means block 0/1. It is not introduced on a specific later block.

Here we state that we change the maximum value defined by EIP-2681.